modules/ting_smart_search/ting_smart_search.module

```diff
@@ -0,0 +1,661 @@
+            }
+            if ($date_delta == 0 && $date_start < time()) {
+              $status = 'active';
+            }
+          }
+          if ($status == 'active') {
+            $smart_search_check_replaced = '';
+            $redirect = $smart_search_object->field_alternate_search_string->value();
+            // Add a general quoted free text search.
+            if (!empty($free_text_query = $redirect)) {
+              $cqlDoctor = new TingSearchCqlDoctor($free_text_query);
+              $smart_search_check_replaced = $cqlDoctor->string_to_cql();
+            }
+
+            // Check for sorting.
+            $smart_search_request_sort = $request->getSort();
+            $params = $_POST;
+
+            if ($smart_search_request_sort == 'rank_frequency') {
+
+              if (array_key_exists('sort', $params)) {
+                // Do nothing here.
+              }
+              else {
+                // Override sort when no sort has been specified.
+                // ting_smart_search_check_sort() - now obsolete.
+                $smart_search_check_sort = $smart_search_object->field_sss_sort->value();
+
+                if (!empty($smart_search_check_sort)) {
+                  $request->setSort($smart_search_check_sort);
+                }
+              }
+
+              $smart_search_check_boosts = ting_smart_search_check_boosts($smart_search_object);
+            }
+
+            if ($smart_search_check_replaced == '') {
+              $query = $request->getQuery();
+              // Probably nothing here.
+            }
+            else {
+              $query = $smart_search_check_replaced;
+            }
+
+            if ($smart_search_check_boosts) {
+              $uboosts = $smart_search_userdefined_boosts;
+              foreach ($smart_search_check_boosts as $boost_field) {
+                $uboosts[] = array(
+                  'fieldName' => $boost_field['field_name'],
+                  'fieldValue' => $boost_field['field_value'],
+                  'weight' => $boost_field['weight'],
+                );
+              }
+              $request->userDefinedBoost = $uboosts;
+              $smart_search_use_auto = false;
+            }
+            $request->queryDebug = 'true';
+            $request->setQuery($query);
+          }
+        }
+      }
+
+      if ($smart_search_use_auto && variable_get('ting_smart_search_automatic_active', true)) {
+        $request->userDefinedBoost = ting_smart_search_auto($smart_search_userdefined_boosts, $request->fullTextQuery);
+      }
+
+      break;
+  }
+
+  return NULL;
+}
+
+/**
+ * Automatically puts the most requested objects first in the search results.
+ *
+ * @param array $smart_search_userdefined_boosts
+ *   Previously defined boosts.
+ * @param string $keys
+ *   The search key.
+ *
+ * @return array
+ *   Array containing the boost of most requested objects
+ */
+function ting_smart_search_auto(array $smart_search_userdefined_boosts, $keys) {
```

Comment:
When trying out this module locally I ran into a fatal error because `$smart_search_userdefined_boosts` was not an array. You might need to look into your data handling.